pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -248,16 +253,24 @@ func CreateGCECloud(apiEndpoint, projectID, region, zone string, managedZones []
 		return nil, err
 	}
 
-	if apiEndpoint != "" {
-		service.BasePath = fmt.Sprintf("%sprojects/", apiEndpoint)
+	client, err = newOauthClient(tokenSource)
+	serviceBeta, err := computebeta.New(client)
+	if err != nil {
+		return nil, err
 	}
 
 	client, err = newOauthClient(tokenSource)
-	serviceBeta, err := computebeta.New(client)
+	serviceAlpha, err := computealpha.New(client)
 	if err != nil {
 		return nil, err
 	}
 
+	if apiEndpoint != "" {
+		service.BasePath = fmt.Sprintf("%sprojects/", apiEndpoint)
+		serviceBeta.BasePath = fmt.Sprintf("%sprojects/", strings.Replace(apiEndpoint, "v1", "beta", 0))
+		serviceAlpha.BasePath = fmt.Sprintf("%sprojects/", strings.Replace(apiEndpoint, "v1", "alpha", 0))
```

Comment:
I was suggesting adding to the user-viewable side: https://github.com/kubernetes/kubernetes/pull/47881/files#diff-e00287b5331f6d39959423afadb6b368R207.